docs/reference/services/ext_authz.md

```diff
@@ -0,0 +1,85 @@
+## The `ext_authz` Protocol
+
+By design, the `ext_authz` protocol used by [the `AuthService`](../auth-service) and by [`External` `Filters`](../../filter-reference) is highly flexible. The authentication service is the first external service invoked on an incoming request (e.g., it runs before the rate limit filter). Because the logic of authentication is encapsulated in an external service, you can use this to support a wide variety of use cases. For example:
+
+* Supporting traditional SSO authentication protocols, e.g., OAuth, OpenID Connect, etc.
+* Support HTTP basic authentication (sample implementation available [here](https://github.com/datawire/ambassador-auth-httpbasic)).
+* Only authenticating requests that are under a rate limit, and rejecting authentication requests above the rate limit.
+* Authenticating specific services (URLs), and not others.
+
+For each request, the external auth service may either
+ 1. return a direct HTTP *response*, intended to be sent back to the requesting HTTP client (normally *denying* the request from being forwarded to the upstream backend service); or
+ 2. return a modification to make to the HTTP *request* before sending it to the upstream backend service (normally *allowing* the request to be forwarded to the upstream backend service with modifications).
+
+The external auth service receives information about every request through Ambassador and must indicate whether the request is to be allowed, or not.  If not, the external auth service provides the HTTP response which is to be handed back to the client.  A potential control flow for Authentication is shown below.
```

Comment:
as a general rule, if you're referencing something, don't say above or below. please include a name of the section; if the section does not have a name, give it one. :D

Reply:
or if this is referring to the image, ignore the bit (but take the advice?)

Reply:
It's referring to the image.